docs/guide/readme.md

```diff
@@ -0,0 +1,269 @@
+
+```html
+<div class="text-center sm:text-left md:text-right"></div>
+```
+
+For behaviors related to a **specific breakpoint** it is recommended to undo this behavior by using a utility class that reapplies the basic behavior on a larger breakpoint.
+
+```html
+<div class="bg-teal md:bg-red lg:bg-teal"></div>
+```
+
+### Custom utility classes
+
+#### First method: Add it in the SCSS files
+
+To manage the import of custom utility classes on Tailwind using a preprocessor you should place them in separate files and import them after Tailwind's utilities imports to avoid specificity issues.
+
+```scss
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+@import "./custom-utilities";
+```
+
+To create custom utility classes that can be used with the breakpoints defined in our configuration you should declare these utility classes in a `@responsive` directive.
+
+```scss
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+@responsive {
+  @variants hover, focus {
+    .rotate-0 {
+      transform: rotate(0deg);
+    }
+    .rotate-90 {
+      transform: rotate(90deg);
+    }
+    .rotate-180 {
+      transform: rotate(180deg);
+    }
+    .rotate-270 {
+      transform: rotate(270deg);
+    }
+  }
+}
+```
+
+The responsive versions of these utility classes will be automatically generated by Tailwind using the breakpoints entered in the configuration file.
+
+```html
+<div class="rotate-0 hover:rotate-90"></div>
+```
+
+You can read the documentation on [variants management](https://tailwindcss.com/docs/pseudo-class-variants/) for more detailed informations.
+
+#### Second method: By creating a plugin
+
+We can add custom utilities with plugin. The following example generates the same classes as in the first method:
+
+```
+// tailwind.config.js
+module.exports = {
+  plugins: [
+    function({ addUtilities }) {
+      const newUtilities = {
+        '.rotate-0': {
+          transform: 'rotate(0deg)',
+        },
+        '.rotate-90': {
+          transform: 'rotate(90deg)',
+        },
+        '.rotate-180': {
+          transform: 'rotate(180deg)',
+        },
+        '.rotate-270': {
+          transform: 'rotate(270deg)',
+        },
+      }
+
+      addUtilities(newUtilities, ['responsive', 'hover'])
+    }
+  ]
+}
+```
+
+Read the [documentation](https://tailwindcss.com/docs/plugins/#adding-utilities) for more informations on plugins.
+
+### Directives and Functions
+
+#### [@apply](https://tailwindcss.com/docs/functions-and-directives#apply)
+
+Apply is useful for [component extraction](https://tailwindcss.com/docs/extracting-components/) when a pattern repeats itself. For buttons for example, it allows to centralize the properties of a component.
+
+These declarations can be used alongside classic CSS declarations.
+
+```scss
+.btn {
+  @apply font-bold py-2 px-4 rounded;
+}
+.btn-blue {
+  @apply bg-blue-500 text-white;
```

Comment:
i do not know if there is a solution for this, but the syntax color is not working well in the preview cause of `@apply` directive :/

Reply:
I don't think we can fix this easily 😞